ltclient/server.js

```diff
@@ -0,0 +1,30 @@
+var webpack = require('webpack');
+var express = require('express');
+var webpackDevMiddleware = require('webpack-dev-middleware');
+var webpackHotMiddleware = require('webpack-hot-middleware');
+var config = require('./webpack.config');
+
+var app = express();
+var port = process.argv[2] ? process.argv[2] - 0 : 4000;
+config.entry.app.unshift(
+  'webpack-hot-middleware/client'
+);
+
+var compiler = webpack(config);
+app.use(webpackDevMiddleware(compiler, {
+  noInfo: true,
+  publicPath: config.output.publicPath
+}));
+app.use(webpackHotMiddleware(compiler));
+
+app.get("/*", function(req, res) {
```

Comment:
Let's say the url of our app is `example.com`. Since it's a SPA, it have many sub pages/routes, e.g. `example.com/signup`. But we are using a single `index.html` for all our pages. So we need to catch user request to a url like `example.com/signup` in server side, and feed it with `index.html`.

See [detal](https://github.com/reactjs/react-router/blob/master/docs/guides/Histories.md#configuring-your-server)

If you use `hashHistory` from `react-router`, you don't need to do this. But you will end up with a very annoying url like `example.com/#/signup?_k=djkeh`. I'm using `browserHistory`, so the "catching" is needed.

See detail in react-router [doc](https://github.com/reactjs/react-router/blob/master/docs/guides/Histories.md#configuring-your-server)
See also one of my [comment](http://react-china.org/t/react-router-hashhistory-browserhistory/6799/3) on react-china.org

@ChopperLee2011 thanks for reviewing.